packages/datetime/src/dateRangePicker.tsx

```diff
@@ -197,13 +215,22 @@ export class DateRangePicker extends AbstractComponent<IDateRangePickerProps, ID
 
     public render() {
         const modifiers = combineModifiers(this.modifiers, this.props.modifiers);
-        const { className, contiguousCalendarMonths, locale, localeUtils, maxDate, minDate } = this.props;
+        const {
+            className,
+            contiguousCalendarMonths,
+            dayPickerProps,
+            locale,
+            localeUtils,
+            maxDate,
+            minDate,
+        } = this.props;
         const isShowingOneMonth = DateUtils.areSameMonth(this.props.minDate, this.props.maxDate);
 
         const { leftView, rightView } = this.state;
         const disabledDays = [{ before: this.props.minDate }, { after: this.props.maxDate }];
 
-        const dayPickerBaseProps: DayPicker.Props = {
+        const dayPickerBaseProps: ReactDayPicker.Props = {
+            ...dayPickerProps,
             disabledDays,
             locale,
```

Comment:
Hmmmmm, I'm not sure the ordering of props is respected here. We'll probably want to use two objects around the `dayPickerProps` and spread them in a specific order:

```tsx

// props in here can be provided either via the top-level API or via the
// `dayPickerProps` object. give priority to the latter, so users can
// pass all react-day-picker modifiers in the single object if desired.
const overridableBaseProps: ReactDayPicker.Props = {
    locale,
    localeUtils,
    modifiers,
};

// these props will be spread *after* the `dayPickerProps` to ensure
// they won't be overridden.
const nonOverridableBaseProps: ReactDayPicker.Props = {
    disabledDays,
    onDayClick: this.handleDayClick,
    onDayMouseEnter: this.handleDayMouseEnter,
    onDayMouseLeave: this.handleDayMouseLeave,
    selectedDays: this.state.value,
};

// ...

// in every case, include all three spreads to guarantee prop order:
<ReactDayPicker
    {...overridableBaseProps}
    {...dayPickerProps} // from this.props
    {...nonOverridableBaseProps}
/>
```

We'll also want the same tests for `DateRangePicker` that we have for `DatePicker` to enforce the ordering of these spreads.

Reply:
you are correct.
i initially implemented it so blueprint locale/localeUtils/modifier take precedence over dayPicker props. since we changed that, we have to adjust the ordering to match the one from [DatePicker](https://github.com/palantir/blueprint/pull/1647/files#diff-619fdabbeb36a1d315ac512931983761R134). thats what i have done. your version is more self-documenting, but also adds more lines of code and adds another level of indirection. It would also have to be written like that in DatePicker imo.